jib-core/src/main/java/com/google/cloud/tools/jib/registry/AbstractManifestPuller.java

```diff
@@ -145,15 +147,29 @@ private T getManifestTemplateFromJson(String jsonString)
     }
     if (schemaVersion == 2) {
       // 'schemaVersion' of 2 can be either Docker V2.2 or OCI.
-      String mediaType = node.get("mediaType").asText();
-      if (V22ManifestTemplate.MANIFEST_MEDIA_TYPE.equals(mediaType)) {
-        return manifestTemplateClass.cast(
-            JsonTemplateMapper.readJson(jsonString, V22ManifestTemplate.class));
+      JsonNode mediaTypeNode = node.get("mediaType");
+      if (mediaTypeNode == null) { // not Docker, hence OCI
+        if (node.get("manifests") != null) {
+          return manifestTemplateClass.cast(
+              JsonTemplateMapper.readJson(jsonString, OciIndexTemplate.class));
```

Comment:
Do we even accept OciIndexTemplate?

Reply:
Not yet ( `getAccept()` doesn't list OCI index). But I'm making this part future proof.